src/main/kotlin/ecommerce/stripe/StripeClient.kt

```diff
@@ -0,0 +1,91 @@
+                "currency=${req.currency}",
+                "payment_method=${req.paymentMethod}",
+                "confirm=true",
+                "automatic_payment_methods[enabled]=true",
+                "automatic_payment_methods[allow_redirects]=never",
+            ).joinToString("&")
+
+        return try {
+            val response =
+                restClient.post()
+                    .uri("https://api.stripe.com/v1/payment_intents")
+                    .header(HttpHeaders.AUTHORIZATION, "Bearer ${stripeProperties.secretKey}")
+                    .contentType(MediaType.APPLICATION_FORM_URLENCODED)
+                    .body(body)
+                    .retrieve()
+                    .toEntity(PaymentResponse::class.java)
+
+            response.body ?: throw IllegalStateException("Empty response from Stripe")
+        } catch (e: RestClientResponseException) {
+            val errorJson = e.responseBodyAsString
+            val stripeError = parseStripeError(errorJson)
+            PaymentResponse(
+                id = "",
+                amount = req.amount,
+                status = "FAILED",
+                clientSecret = null,
+                errorMessage = convertMessageFromError(stripeError),
+            )
+        } catch (e: Exception) {
+            throw IllegalArgumentException("Stripe error: ${e.message}", e)
+        }
+    }
+
+    private fun parseStripeError(json: String?): PaymentErrorResponse? {
+        if (json.isNullOrBlank()) return null
+        return try {
+            val root = jacksonObjectMapper().readTree(json)
+            println(root.toString())
+            val errNode = root.path("error")
+            PaymentErrorResponse(
+                code = errNode.path("code").asText(null),
+                // converts value to a string or provides null if absent/missing.
+                declineCode = errNode.path("decline_code").asText(null),
+                message = errNode.path("message").asText(null),
+            )
+        } catch (ex: Exception) {
+            null
+        }
+    }
+
+    private fun convertMessageFromError(error: PaymentErrorResponse?): String {
+        return when (error?.code) {
+            "card_declined" ->
+                when (error.declineCode) {
+                    "insufficient_funds" -> "Payment failed: insufficient balance."
+                    "expired_card" -> "Payment failed: your card has expired."
+                    "incorrect_cvc" -> "Payment failed: incorrect CVC."
+                    "lost_card" -> "Payment failed: lost card."
+                    "stolen_card" -> "Payment failed: stolen card."
+                    else -> "Payment was declined by your bank."
+                }
+            "expired_card" -> "Payment failed: your card has expired."
+            "processing_error" -> "Payment failed due to a processing error."
+            else -> error?.message ?: "Payment failed for an unknown reason."
+        }
+    }
+}
```

Comment:
I think we do need a way to handle error messages and convert it to our own, so I do agree that this process is necessary.

However, just like our previous review, `PaymentErrorResponse` seems like it should be open to situations where it handles more than Stripe's api response. So rather than using `PaymentErrorResponse` to convert error message, I think this should be done in Stripe related logic before we use the general `PaymentErrorResponse`.

Or to just make it more simple, change the name to `StripePaymentErrorResponse` :)
Since `PaymentErrorResponse` is only used in `StripeClient`, I think this way of refactoring seems okay. But to make it better, it'll be great to have Stripe related package and manage all the related classes there.